model/src/lib.rs

```diff
@@ -51,35 +52,94 @@ pub struct UserOrder {
     pub signature: Signature,
 }
 
-impl UserOrder {
+impl OrderCreation {
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
 }
 
-/// An order that is returned when querying the orderbook.
-///
-/// Contains extra fields thats are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+// uid as 56 bytes: 32 for orderDigest, 20 for ownerAddress and 4 for validTo
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct OrderUid(pub [u8; 56]);
+
+impl Default for OrderUid {
+    fn default() -> Self {
+        OrderUid([0 as u8; 56])
+    }
+}
+
+impl Serialize for OrderUid {
```

Comment:
I used this crate the other day and was very happy with it: https://docs.rs/serde_with/1.6.0/serde_with/hex/struct.Hex.html. It has a bunch of helpers for implementing serialization and deserialization.

Alternatively, you could implement `Display` and `FromStr` and then use: https://docs.rs/serde_with/1.6.0/serde_with/struct.DisplayFromStr.html. This way you can customize how the hex is serialized but don't have to add the complex `Visitor` code.

Reply:
I tried your first suggestion, but it does only work for vectors or arrays below the size of 32 :( Even with the new compiler.

I also tried your second suggestion. But here I would have to implement the `Display` trait for [u8;56] and not for OrderUid. Otherwise, I get:
```
`[u8; 56]` doesn't implement `std::fmt::Display`
```
And I am not allowed to implement Display for [0;56], as I am not in the "defining" crate.

Somehow my knowledge is not sufficient to jump over these hurdles.